### PR TITLE
Note about OnBuild STI strategy

### PR DESCRIPTION
**Pull request by mfojtik**

### CODE REVIEW

**Comment by mfojtik** [comment]
@CowboysFan @bparees @soltysh can you review this please?

File: image_writers_guide/sti.adoc [hunk 1]
```diff
@@ -32,7 +32,37 @@ image::sti-flow.png[STI workflow]
...
+The ONBUILD instructions can be found in many "official" Docker images. For example:
+
+https://registry.hub.docker.com/u/library/ruby
+https://registry.hub.docker.com/u/library/node
+https://registry.hub.docker.com/u/library/python
+
+For more details about how ONBUILD works, visit
+https://docs.docker.com/reference/builder/#onbuild.
+
+The STI has a different strategy when the Docker image with ONBUILD instructions
```
Comment (soltysh) [inline image_writers_guide/sti.adoc]: when ~~the~~ Docker image

File: using_openshift/builds.adoc [hunk 2]
```diff
@@ -46,6 +46,32 @@ Builds can be manually canceled using the following command:
...
+              "uri": "git://github.com/openshift/ruby-hello-world.git"
+            },
+            "contextDir": "app/dir",
+          },
+        }
+
+The 'type' field describe what SCM will be used to fetch your source code. In this
+example, your source code lives in 'GIT' repository. Then the 'git' field
+contains the 'uri' to your remote GIT repository. It might optionally specify
+the 'ref' field if you want to checkout a specific GIT ref.
```
Comment (soltysh) [inline using_openshift/builds.adoc]: Maybe short info/link to what git refs are?

File: using_openshift/builds.adoc [hunk 3]
```diff
@@ -46,6 +46,32 @@ Builds can be manually canceled using the following command:
...
+
+The 'contextDir' field allows you to override the default location inside the
+source code repository, where the build will lookup for the application source
```
Comment (soltysh) [inline using_openshift/builds.adoc]: ...build will look~~up~~ for the application...

**Comment by soltysh** [comment]
Small nits but generally LGTM.

**Comment by mfojtik** [comment]
@soltysh thanks! all fixed

**Comment by adellape** [comment]
Reviewed, LGTM. Thx!

**Pull request merged**